src/setting.rs

```diff
@@ -201,636 +182,271 @@ pub enum SettingData {
-        let (input, value) = nom::number::complete::u8(input)?;
-        let value = match value {
-            0x80 => Self::Unlock,
-            0x81 => Self::Lock,
-            _ => Self::Unknown(value),
-        };
-        Ok((input, value))
-    }
 }
 
 /// Found at "PLAYER > DJ SETTING > NEEDLE LOCK" of the "My Settings" page in the Rekordbox
 /// preferences.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
+#[binrw]
+#[brw(repr = u8)]
 pub enum NeedleLock {
     /// Named "UNLOCK" in the Rekordbox preferences.
-    Unlock,
+    Unlock = 0x80,
     /// Named "LOCK" in the Rekordbox preferences.
     Lock,
-    /// Unknown value.
-    Unknown(u8),
-}
-
-impl NeedleLock {
-    fn parse(input: &[u8]) -> IResult<&[u8], Self> {
-        let (input, value) = nom::number::complete::u8(input)?;
-        let value = match value {
-            0x80 => Self::Unlock,
-            0x81 => Self::Lock,
-            _ => Self::Unknown(value),
-        };
-        Ok((input, value))
-    }
 }
 
 /// Found at "PLAYER > DJ SETTING > QUANTIZE BEAT VALUE" of the "My Settings" page in the Rekordbox
 /// preferences.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
+#[binrw]
+#[brw(repr = u8)]
 pub enum QuantizeBeatValue {
     /// Named "1/8 Beat" in the Rekordbox preferences.
-    EighthBeat,
+    EighthBeat = 0x83,
     /// Named "1/4 Beat" in the Rekordbox preferences.
-    QuarterBeat,
+    QuarterBeat = 0x82,
     /// Named "1/2 Beat" in the Rekordbox preferences.
-    HalfBeat,
+    HalfBeat = 0x81,
     /// Named "1 Beat" in the Rekordbox preferences.
-    FullBeat,
-    /// Unknown value.
-    Unknown(u8),
-}
-
-impl QuantizeBeatValue {
-    fn parse(input: &[u8]) -> IResult<&[u8], Self> {
-        let (input, value) = nom::number::complete::u8(input)?;
-        let value = match value {
-            0x80 => Self::FullBeat,
-            0x81 => Self::HalfBeat,
-            0x82 => Self::QuarterBeat,
-            0x83 => Self::EighthBeat,
-            _ => Self::Unknown(value),
-        };
-        Ok((input, value))
-    }
+    FullBeat = 0x80,
 }
 
 /// Found at "PLAYER > DJ SETTING > HOT CUE AUTO LOAD" of the "My Settings" page in the Rekordbox
 /// preferences.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
+#[binrw]
+#[brw(repr = u8)]
 pub enum HotCueAutoLoad {
     /// Named "OFF" in the Rekordbox preferences.
-    Off,
+    Off = 0x80,
     /// Named "rekordbox SETTING" in the Rekordbox preferences.
-    RekordboxSetting,
+    RekordboxSetting = 0x82,
     /// Named "On" in the Rekordbox preferences.
-    On,
-    /// Unknown value.
-    Unknown(u8),
-}
-
-impl HotCueAutoLoad {
-    fn parse(input: &[u8]) -> IResult<&[u8], Self> {
-        let (input, value) = nom::number::complete::u8(input)?;
-        let value = match value {
-            0x80 => Self::Off,
-            0x81 => Self::On,
-            0x82 => Self::RekordboxSetting,
-            _ => Self::Unknown(value),
-        };
-        Ok((input, value))
-    }
+    On = 0x81,
```

Comment:
consider reordering

Reply:
At the moment the order is the same as in the rekordbox preferences. We'd lose that info if we reorder.

Reply:
Mhmm. IMO ordering it `ON`, `OFF`, `RekordboxSetting` makes more sense semantically though. I don't think there is much value in retaining the same order as in the rekordbox preferences. It doesn't make a difference semantically and I also don't see a reason for keeping that ordering for documentation purposes. After all, the (imo) weird ordering of these option in the UI, could just be a UI bug.

Reply:
I can't really judge if this is an UI bug or not, but this is how it looks in RB6:

![image](https://user-images.githubusercontent.com/1834516/155426213-02b21608-aaf2-4a5c-acd3-930dc7b6c048.png)

I don't think it hurts readability that much, let's just keep it for now.